src/utils/extract-date-parts-from-date-string.ts

```diff
@@ -7,10 +7,25 @@ export function extractDatePartsFromDateString(dateString: string) {
         throw new Error(`Date string is invalid, received "${dateString}"`)
     }
 
-    const [, yearStr, , monthStr, , dayStr] = parts
+    let yearStr, monthStr, dayStr, format
+
+    if (parts[1]) {
```

Comment:
what does this check exactly? what is `parts[1]` in each case

Reply:
@kabaros I initially wrote it like [this](https://github.com/dhis2/multi-calendar-dates/blob/114d9e3baf904ec60534d8718d8d5b74b29cd7b5/src/utils/extract-date-parts-from-date-string.ts), but it caused issues with Prettier, it adds `;` at the beginning of these lines.

`[, yearStr, , monthStr, , dayStr] = parts;`
 `[, , , , , , dayStr, , monthStr, , yearStr] = parts;`

Parts[1] equal to this part (\d{4}), which means the format provided starts with 4 numbers.

Reply:
ok .. let's have it like this for now. We can see if it can be refactored later.